clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java

```diff
@@ -690,22 +691,22 @@ private boolean ignoreNewUpdate(
         if (updateOperationTimestamp >= valueLevelTimestamp) {
           return false;
         }
-        toUpdateFieldNames = WriteComputeSchemaConverter.getNamesOfFieldsToBeUpdated(writeComputeRecord);
-        for (String toUpdateFieldName: toUpdateFieldNames) {
-          if (oldValueSchema.getField(toUpdateFieldName) == null) {
+        for (Schema.Field field: writeComputeRecord.getSchema().getFields()) {
+          Object fieldObject = writeComputeRecord.get(field.name());
+          if (getFieldOperationType(fieldObject) != NO_OP_ON_FIELD && oldValueSchema.getField(field.name()) == null) {
             return false; // Write Compute tries to update a non-existing field in the old value (schema).
           }
         }
         return true; // Write Compute does not try to update any non-existing fields in the old value (schema).
 
       case PER_FIELD_TIMESTAMP:
         GenericRecord timestampRecord = (GenericRecord) oldTimestampObject;
-        toUpdateFieldNames = WriteComputeSchemaConverter.getNamesOfFieldsToBeUpdated(writeComputeRecord);
-        for (String toUpdateFieldName: toUpdateFieldNames) {
-          if (timestampRecord.get(toUpdateFieldName) == null) {
+        for (Schema.Field field: writeComputeRecord.getSchema().getFields()) {
+          Object fieldObject = writeComputeRecord.get(field.name());
```

Comment:
Since the field comes from the schema of the record your are accessing, there is no need to access by name. You can directly use the field’s position field instead, which is more efficient (and which is what ends up getting used anyway). Same in the other case, above.

Note that we cannot blindly use field positions everywhere, for example if the record we are accessing has a different schema than the one where the field object comes from.

Reply:
Sure, I can access them by `field.pos`. But what if in future write compute schema field order gets modified, then it could fail ?

Reply:
yeah, we could potentially do one time validation of schema->field order match and cache that and use that info to get field by pos or name. But sounds bit round-about and may not be worth it.

Reply:
I am wondering at this point all there GR should be deserialized by the same schema version right? Either it is a superset schema version or a latest schema version (this could happen), but WC schema / RMD schema should match?

Reply:
I would say: use `pos()` only when the field comes from the same record you're going to `get(pos)` on. For any other case, use `name()`.

Reply:
Yeah okay...for safety this is probably better.